ql/instruments/zerocouponswap.hpp

```diff
@@ -0,0 +1,135 @@
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+ */
+
+/*! \file zerocouponswap.hpp
+ \brief Zero-coupon interest rate swap
+ */
+
+#ifndef quantlib_zerocouponswap_hpp
+#define quantlib_zerocouponswap_hpp
+
+#include <ql/instruments/swap.hpp>
+#include <ql/time/calendar.hpp>
+#include <ql/time/daycounter.hpp>
+#include <ql/cashflows/rateaveraging.hpp>
+
+namespace QuantLib {
+    class IborIndex;
+
+    //! Zero-coupon interest rate swap
+    /*! Quoted in terms of a known fixed cash flow \f$ N^{FIX} \f$ or
+        a fixed rate \f$ K \f$, where:
+        \f[
+        N^{FIX} = N \left[ (1+K)^{\alpha(T_{0}, T_{K})}-1 \right] ,
+        \f]
+        with \f$ \alpha(T_{0}, T_{K}) \f$ being the time fraction
+        between the start date of the contract \f$ T_{0} \f$ and
+        the end date \f$ T_{K} \f$ - according to a given day count
+        convention. \f$ N \f$ is the base notional amount prior to
+        compounding.
+        The floating leg also pays a single cash flow \f$ N^{FLT} \f$,
+        which value is determined by periodically averaging (e.g. every
+        6 months) interest rate index fixings.
+        Assuming the use of compounded averaging the projected value of
+        the floating leg becomes:
+        \f[
+        N^{FLT} = N \left[ \prod_{k=0}^{K} (1+\alpha(T_{k},T_{k+1}) 
+                           L(T_{k},T_{k+1})) -1 \right],
+        \f]
+        where \f$ L(T_{i}, T_{j})) \f$ are interest rate index fixings
+        for accrual period \f$ [T_{i}, T_{j}] \f$.
+        For simple averaging we have:
+        \f[
+        N^{FLT} = N \left[ \sum_{k=0}^{K} \alpha(T_{k},T_{k+1})
+                           L(T_{k},T_{k+1}) \right].
+        \f]
+        For a par contract, it holds that:
+        \f[
+        P_n(0,T) N^{FIX} = P_n(0,T) N^{FLT}
+        \f]
+        where \f$ T \f$ is the final payment time, \f$ P_n(0,t) \f$
+        is the nominal discount factor at time \f$ t \f$.
+
+        At maturity the two single cashflows are swapped.
+
+        \note we do not need Schedules on the legs because they use
+              one or two dates only per leg. Those dates are not 
+              adjusted for potential non-business days. Only the
+              payment date is subject to adjustment.
+    */
+
+    class ZeroCouponSwap : public Swap {
+      public:
+        enum Type { Receiver = -1, Payer = 1 };
+        
+        ZeroCouponSwap(Type type,
+                       Real baseNominal,
+                       const Date& startDate,
+                       const Date& maturityDate, 
+                       Real fixedPayment,
+                       ext::shared_ptr<IborIndex> iborIndex,
+                       const Calendar& paymentCalendar,
+                       BusinessDayConvention paymentConvention = Following,
+                       Natural paymentDelay = 0,
+                       RateAveraging::Type averagingMethod = RateAveraging::Compound);
```

Comment:
Do swaps actually exist with simple averaging instead of compound?

Reply:
Quite frankly, not to my knowledge. Usually, zero coupon swaps are compounded. It was just easy to expose both possibilities. I could remove the option to create an instrument with simple averaging. Let me know, please.

Reply:
Yes, I'd keep it simple and remove the option.  We can always add it if needed.

Reply:
Ok, I removed simple averaging.